src/emulator/storage/rules/manager.ts

```diff
@@ -100,13 +131,50 @@ export class StorageRulesManager {
   }
 }
 
-function readSourceFile(fileName: string): string {
-  try {
-    return fs.readFileSync(fileName).toString();
-  } catch (error: any) {
-    if (error.code === "ENOENT") {
-      throw new FirebaseError(`File not found: ${fileName}`);
+/**
+ * Maintains a mapping from storage resource to {@link StorageRulesManagerImplementation} and
+ * directs calls to the appropriate instance.
+ */
+class StorageRulesManagerRegistry {
```

Comment:
What's the difference between a `StorageRulesManagerImplementation` and a `StorageRulesManagerRegistry` with a single map entry? If it's just the ability to maintain a default security rule, you can build that into the registry and get rid of the `interface StorageRulesManger` entirely.

Reply:
Talked offline; will go ahead with `StorageRulesManger` interface